app/forms/ip-pool-edit.tsx

```diff
@@ -30,23 +30,27 @@ EditIpPoolSideModalForm.loader = async ({ params }: LoaderFunctionArgs) => {
 export function EditIpPoolSideModalForm() {
   const queryClient = useApiQueryClient()
   const navigate = useNavigate()
-
   const poolSelector = useIpPoolSelector()
 
-  const onDismiss = () => navigate(pb.ipPools())
-
   const { data: pool } = usePrefetchedApiQuery('ipPoolView', { path: poolSelector })
 
+  const form = useForm({ defaultValues: pool })
+  const onDismiss = () => navigate(pb.ipPool({ pool: poolSelector.pool }))
+
   const editPool = useApiMutation('ipPoolUpdate', {
     onSuccess(_pool) {
       queryClient.invalidateQueries('ipPoolList')
+      if (pool.name !== _pool.name) {
+        // as the pool's name has changed, we need to navigate to an updated URL
+        navigate(pb.ipPool({ pool: _pool.name }))
+      } else {
+        queryClient.invalidateQueries('ipPoolView')
+        onDismiss()
+      }
```

Comment:
I think this is equivalent.

```suggestion
      queryClient.invalidateQueries('ipPoolList')
      if (pool.name === _pool.name) {
        queryClient.invalidateQueries('ipPoolView')
      }
      navigate(pb.ipPool({ pool: _pool.name }))
```

Is the reason for the conditional that you were getting that error flash on nav? Otherwise this would work, wouldn't it?

```ts
queryClient.invalidateQueries('ipPoolList')
queryClient.invalidateQueries('ipPoolView')
navigate(pb.ipPool({ pool: _pool.name }))
```

Or maybe navigate first I guess?